<?xml version="1.0" encoding="utf-8"?>

<launch>
    <param name="/use_sime_time" value="true"/>

    <!-- Preprocessing pointcloud and image -->
    <param name="preprocessing_image_topic_sub" value="/zed2i/zed_node/left/image_rect_color" />
    <param name="preprocessing_depth_topic_sub" value="/zed2i/zed_node/point_cloud/cloud_registered" />
    <param name="preprocessing_image_topic_pub" value="/zed2i/zed_node/preprocessed/left/image_rect_color" />
    <param name="preprocessing_depth_topic_pub" value="/zed2i/zed_node/preprocessed/point_cloud/cloud_registered" />

    <node name="ground_removal_node"  pkg="ground_removal" type="ground_removal_node" output="screen" />

    <!-- #### YOLOv3 - 2D Object Detection #### -->
    <arg name="image" value="/zed2i/zed_node/preprocessed/left/image_rect_color" />
    <arg name="network_param_file" default="$(find darknet_ros)/config/yolov3.yaml" />
    <include file="$(find darknet_ros)/launch/yolo_v3.launch">
        <arg name="image" value="$(arg image)" />
    </include>

    <!-- #### Frustum Pointnet - 3D Object Detection #### -->
    <remap from="/bounding_box_2d_publisher" to="/darknet_ros/bounding_boxes" />
    <remap from="/gt_bb_publisher_3D"        to="/gt_bb_publisher_3D" />
    <node name="fpointnet_subscriber" pkg="frustum_pointnet" type="kitti_sub_node" output="screen" />

    <!-- #### TF static transform publisher #### -->
    <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 1 map base_link 100" />

    <!-- #### ZED Image Publisher #### -->
    <!-- <remap from="/image_publisher" to="/zed/zed_node/left/image_rect_color" /> -->
    <node pkg="rosbag" type="play" name="rosbag_play" args="--clock $(arg bag_path) -r 1" required="true" />

</launch>